### PR TITLE
CON-3307-Review Adverts Tile

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -86,7 +86,8 @@
         "ProviderCommitmentsSiteUrl": "",
         "SignProviderAgreementUrl": "",
         "ReservationsSiteUrl": "",
-        "EmployerFavouritesUrl": ""
+        "EmployerFavouritesUrl": "",
+		"TrainingProviderPermissionUrl": ""
```
Comment (cofaulco) [inline azure/template.json:90]: dodgy formatting

File: src/Employer/Employer.Web/Controllers/VacancyManageController.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public async Task<IActionResult> ManageVacancy(VacancyRouteModel vrm)
 
             var vacancy = await _orchestrator.GetVacancy(vrm);
 
-            if (vacancy.CanEdit)
```
Comment (cofaulco) [inline src/Employer/Employer.Web/Controllers/VacancyManageController.cs:36]: do we need to change the name of this property? This is on employer side so the context comes from that

File: src/Employer/Employer.Web/Mappings/VacancyDisplayViewModelMapper.cs [hunk 3]
```diff
@@ -43,6 +43,7 @@ public async Task MapFromVacancyAsync(DisplayVacancyViewModel vm, Vacancy vacanc
             vm.ApplicationUrl = vacancy.ApplicationUrl;
             vm.CanDelete = vacancy.CanDelete;
             vm.CanSubmit = vacancy.CanSubmit;
+            vm.CanReadyForReview = vacancy.CanReadyForReview;
```
Comment (cofaulco) [inline src/Employer/Employer.Web/Mappings/VacancyDisplayViewModelMapper.cs:46]: could this property be named `IsReadyForReview` or `CanReview`

File: src/Employer/Employer.Web/Orchestrators/VacancyViewOrchestrator.cs [hunk 4]
```diff
@@ -72,6 +72,12 @@ public async Task<DisplayVacancyViewModel> GetFullVacancyDisplayViewModelAsync(V
                     await _vacancyDisplayMapper.MapFromVacancyAsync(submittedViewModel, vacancy);
                     submittedViewModel.SubmittedDate = vacancy.SubmittedDate.Value.AsGdsDate();
                     return submittedViewModel;
+                case VacancyStatus.Review:
+                    var reviewViewModel = new ReviewVacancyViewModel();
+                    reviewViewModel.EducationLevelName =
```
Comment (cofaulco) [inline src/Employer/Employer.Web/Orchestrators/VacancyViewOrchestrator.cs:77]: this property set could be within the initialisation of the object:

File: src/Employer/Employer.Web/Orchestrators/VacancyViewOrchestrator.cs [hunk 5]
```diff
@@ -72,6 +72,12 @@ public async Task<DisplayVacancyViewModel> GetFullVacancyDisplayViewModelAsync(V
...
+                        EducationLevelNumberHelper.GetEducationLevelNameOrDefault(programme.EducationLevelNumber, programme.ApprenticeshipLevel);
```
Comment (cofaulco) [inline src/Employer/Employer.Web/Orchestrators/VacancyViewOrchestrator.cs:78]:
```suggestion
                    var reviewViewModel = new ReviewVacancyViewModel{
                    EducationLevelName =
                        EducationLevelNumberHelper.GetEducationLevelNameOrDefault(programme.EducationLevelNumber, programme.ApprenticeshipLevel)
                        };
```

File: azure/template.json (hunk 1 above)
Comment (VasanthaKasirajan3008) [reply azure/template.json]: i know , looks okay in VS code.. i have used jsonformatter  , thats why loads of diff.

File: src/Employer/Employer.Web/Mappings/VacancyDisplayViewModelMapper.cs (hunk 3 above)
Comment (VasanthaKasirajan3008) [reply src/Employer/Employer.Web/Mappings/VacancyDisplayViewModelMapper.cs]: I have changed to CanReview

File: src/Employer/Employer.Web/Orchestrators/VacancyViewOrchestrator.cs (hunk 4 above)
Comment (VasanthaKasirajan3008) [reply src/Employer/Employer.Web/Orchestrators/VacancyViewOrchestrator.cs]: done

File: src/Employer/Employer.Web/Orchestrators/VacancyViewOrchestrator.cs (hunk 5 above)
Comment (VasanthaKasirajan3008) [reply src/Employer/Employer.Web/Orchestrators/VacancyViewOrchestrator.cs]: very nice to see about the suggestion box.. done

File: src/Employer/Employer.Web/ViewModels/Dashboard/DashboardViewModel.cs [hunk 6]
```diff
@@ -18,6 +18,9 @@ public class DashboardViewModel
         public int VacancyCountDraft => Vacancies.Count(v => v.Status == VacancyStatus.Draft);
         public string VacancyTextDraft => "advert".ToQuantity(VacancyCountDraft, ShowQuantityAs.None);
         public bool HasDraftVacancy => VacancyCountDraft > 0;
+        public int VacancyCountReview => Vacancies.Count(v => v.Status == VacancyStatus.Review);
+        public string VacancyTextReview => "advert".ToQuantity(VacancyCountReview, ShowQuantityAs.None);
```
Comment (cofaulco) [inline src/Employer/Employer.Web/ViewModels/Dashboard/DashboardViewModel.cs:22]: could we user the VacancyTerm constant implemented in Pauls branch?

File: src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml [hunk 7]
```diff
@@ -182,6 +190,9 @@
                 <p class="govuk-body">Set up and manage the emails you receive about your adverts and applications.</p>
             </div>
         </esfaFeatureEnabled>
+
+
+
```
Comment (cofaulco) [inline src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml:195]: random whitespace

File: src/Employer/Employer.Web/ViewModels/Dashboard/DashboardViewModel.cs (hunk 6 above)
Comment (VasanthaKasirajan3008) [reply src/Employer/Employer.Web/ViewModels/Dashboard/DashboardViewModel.cs:22]: Paul already added the changes to the employer side https://github.com/SkillsFundingAgency/das-recruit/pull/958/files
do you want me to bring those changes (cherry pick) to this branch

File: src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml (hunk 7 above)
Comment (VasanthaKasirajan3008) [reply src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml]: removed

File: src/Shared/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs [hunk 8]
```diff
@@ -69,11 +69,20 @@ public class Vacancy
                                   Status == VacancyStatus.Referred)
                                  && IsDeleted == false;
         /// <summary>
-        /// We can only edit draft & referred vacancies that have not been deleted
+        /// We can only edit draft , referred vacancies that have not been deleted
```
Comment (cofaulco) [inline src/Shared/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs:72]: why change the '&' to ','?

Comment (VasanthaKasirajan3008) [reply src/Shared/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs:72]: changed the text to add Review status. as we have decided to use CanEmployerEdit ..  will undo the text changes

Comment (VasanthaKasirajan3008) [reply src/Shared/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs]: done

**Review by cofaulco — APPROVED** [review]

**Pull request merged**